barcode/tests/__init__.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+from barcode import *
+from barcode_utils import generate_barcode_from_badge_num, get_badge_num_from_barcode
+
+
+@pytest.fixture
+def cfg(monkeypatch):
+    config['secret'] = dict()
+    config['secret']['barcode_key'] = "ABCDEF1234"
+    config['secret']['barcode_salt'] = 42
+    config['secret']['barcode_event_id'] = 0xFF
+
+
+def test_cfg_fixtures(cfg):
```

Comment:
Sorry for not having time to respond last week.  I realize this has already been merged, but I'm making an effort to review and respond to everything for future reference.

> yea, gotcha. makes sense. actually is there anyplace where we validate config on startup, I could extend that to plugins and put these checks here? @EliAndrewC ?

ConfigObj (the library we use for our config files which provides that handy specfile concept) already has a way to indicate a valid range for integers.  You can just drop this unit test and instead modify your configspec to read something like:

```
barcode_salt = integer(default=0, min=0, max=1000000)
barcode_event_id = integer(default=0, min=0, max=255)
barcode_key = string(min=10, max=10)  # min and max refer to length for strings
```

The length restrictions are strings are from memory, so that might actually have a different syntax.  Relatedly, if those config options are required for the plugin to function, you probably don't want default values, since you probably want an exception to be raised if they're not provided.